### PR TITLE
feat: allow CGO_ENABLED to be configurable for development

### PR DESCRIPTION
**Pull request by malept**
## What this PR does / why we need it

We should not default to compile builds with CGo. This makes it difficult to build a service on a Linux host and run it in a container, particularly if the host and container have different glibc versions. Instead, use `service.yaml` to configure whether CGo compilation is enabled, defaulting to false.

## Jira ID

[DT-4070]

## Notes for your reviewers

A corresponding PR will need to be written for production builds, in `stencil-golang`.

[DT-4070]: https://outreach-io.atlassian.net/browse/DT-4070?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: shell/cgo-enabled.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+# This is a small script to determine whether a service manifest has
+# explicitly enabled compiling with CGo, and print out the appropriate
+# value for the CGO_ENABLED environment variable. Defaults to disabled.
+
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+# shellcheck source=./lib/bootstrap.sh
+source "$DIR/lib/bootstrap.sh"
+
+if [[ "$(yq -r ".arguments.enableCGo" <"$(get_service_yaml)")" == "true" ]]; then
```
Comment (jaredallard) [inline shell/cgo-enabled.sh:10]:
```suggestion
if [[ "$(yq -r ".arguments.enableCGO" <"$(get_service_yaml)")" == "true" ]]; then
```

I think it technically should be `CGO`...

Comment (jaredallard) [inline shell/cgo-enabled.sh:10]: Add field to `manifest.yaml`?

Comment (malept) [reply shell/cgo-enabled.sh:10]: It's unclear to me what the canonical capitalization of it is. It's very inconsistent in https://pkg.go.dev/cmd/cgo.

I guess it's either `Cgo` or `cgo`, so the suggestion should be
```suggestion
if [[ "$(yq -r ".arguments.enableCgo" <"$(get_service_yaml)")" == "true" ]]; then
```

Comment (malept) [reply shell/cgo-enabled.sh]: I swear I did that but I'm not seeing it anymore.

**Review by jaredallard — APPROVED** [review]

**Pull request merged**